src/crimson/os/seastore/transaction_manager.h

```diff
@@ -644,6 +644,93 @@ class TransactionManager : public ExtentCallbackInterface {
     ExtentPlacementManager::dispatch_result_t dispatch_result,
     std::optional<journal_seq_t> seq_to_trim = std::nullopt);
 
+  /**
+   * alloc_existing_extent
+   *
+   * Allocates a new extent at given existing_paddr that must be absolute and 
+   * use the buffer to fill the new extent if buffer exists. Otherwise, will
+   * not read disk to fill the new extent.
+   * Returns the new extent.
+   */
+  using alloc_existing_extent_iertr =
+    alloc_extent_iertr::extend_ertr<Device::read_ertr>;
+  template <typename T>
+  using alloc_existing_extent_ret =
+    alloc_existing_extent_iertr::future<TCachedExtentRef<T>>;
+  template <typename T>
+  alloc_existing_extent_ret<T> alloc_existing_extent(
+    Transaction &t,
+    laddr_t existing_laddr,
+    paddr_t existing_paddr,
+    extent_len_t length,
+    laddr_t laddr,
+    std::optional<ceph::bufferptr> bp) {
+    LOG_PREFIX(TransactionManager::alloc_existing_extent);
+    SUBDEBUG(seastore_tm, "alloc existing extent: existing_laddr: {}, "
+      "existing_paddr: {}, length: {}, has data in cache: {} ",
+      existing_laddr, existing_paddr, length, bp.has_value()? "true" : "false");
+    assert(existing_laddr == laddr ||
+      (existing_laddr - laddr + length == bp->length()));
```

Comment:
checking remap_laddr/length is too strict in transaction_manager, making sure that the remap range is within the original_range should be sufficient.